crates/zizmor/src/main.rs

```diff
@@ -142,6 +144,101 @@ struct App {
     inputs: Vec<String>,
 }
 
+/// Shell with auto-generated completion script available.
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+#[non_exhaustive]
+pub enum Shell {
+    /// Bourne Again `SHell` (bash)
+    Bash,
+    /// Carapace multi-shell completion engine
+    Carapace,
+    /// Elvish shell
+    Elvish,
+    /// Friendly Interactive `SHell` (fish)
+    Fish,
+    /// Nushell
+    Nushell,
+    /// `PowerShell`
+    PowerShell,
+    /// Z `SHell` (zsh)
+    Zsh,
+}
+
+impl Display for Shell {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.to_possible_value()
+            .expect("no values are skipped")
+            .get_name()
+            .fmt(f)
+    }
+}
+
+impl FromStr for Shell {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        for variant in Self::value_variants() {
+            if variant.to_possible_value().unwrap().matches(s, false) {
+                return Ok(*variant);
+            }
+        }
+        Err(format!("invalid variant: {s}"))
+    }
+}
```

Comment:
Apologies if I'm missing it, but where do we this `FromStr` impl?

Reply:
We don't, it was part of the block I cargo-culted over from https://github.com/clap-rs/clap/blob/master/clap_complete/src/aot/shells/shell.rs. It was between two parts that were needed, and I forgot to remove it.